### PR TITLE
Removed csv, switched to using arrays

### LINKED ISSUE #4: Improve parsing of marine_result->output
After profiling the code, it appears that parsing the CSV output takes 9% of the code runtime.

Find a faster way to parse the content (can make it not be CSV if needed)

### PR DESCRIPTION
**Pull request by shaharyarn**
Before:
after_speedup

After:
after_csv

### CODE REVIEW

File: marine/marine.py [hunk 1]
```diff
@@ -167,11 +165,11 @@ def validate_fields(
         return bool(self._marine.validate_fields(fields_c_arr, fields_len))
 
     @staticmethod
-    def _parse_output(output: str) -> List[str]:
-        # TODO: this is a bottleneck. Find a better way to provide output from the c code
-        f = StringIO(output)
-        csv_parsed_output = next(csv.reader(f, delimiter="\t", quotechar='"'), [""])
-        return [value if len(value) > 0 else None for value in csv_parsed_output]
+    def _parse_output(output, amount: int) -> Iterable[str]:
```
Comment (yehonatanz) [inline marine/marine.py:168]: `length`/`size`

File: marine/marine.py [hunk 2]
```diff
@@ -167,11 +165,11 @@ def validate_fields(
...
+        return (
+            output[i].decode("utf=8") if output[i] is not None else None
```
Comment (yehonatanz) [inline marine/marine.py:170]: Didn't you mean `utf-8`?

**Review by yehonatanz — CHANGES_REQUESTED** [review]
Looks great!
Had a couple of notes though.

File: marine/marine.py (hunk 1 above)
Comment (tomlegkov) [reply marine/marine.py:168]: And typing on `output`

**Review by tomlegkov — APPROVED** [review]

**Pull request closed without merging**